sec_certs/dataset/dataset.py

```diff
@@ -82,12 +82,14 @@ def __getitem__(self, item: str):
         return self.certs.__getitem__(item.lower())
 
     def __setitem__(self, key: str, value: 'Certificate'):
-        self.certs.__setitem__(key.lower(), value)
+        dict(self.certs).update({key.lower(): value})
```

Comment:
There is no way this will work right? You are creating a new dictionary object here and updating it, but this will do nothing to the inner `self.certs` object which will remain unchanged.